### PR TITLE
HPCC-32519 ReadtmsAllowClose() handle JSOCKERR_not_opened

### PR DESCRIPTION
**Pull request by mckellyln**
## Type of change:
- [x] This change is a bug fix (non-breaking change which fixes an issue).
- [ ] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [x] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [x] The commit message title makes sense in a changelog, by itself.
  - [x] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [ ] I have read the CONTRIBUTORS document.
- [ ] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [ ] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Cloud-compatibility
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [x] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Smoketest:
- [x] Send notifications about my Pull Request position in Smoketest queue.
- [ ] Test my draft Pull Request.

### CODE REVIEW

**Comment by mckellyln** [comment]
@jakesmith what do you think ?  Could there be other places in the code like this ?

File: common/thorhelper/thorsoapcall.cpp [hunk 1]
```diff
@@ -2082,10 +2082,11 @@ class CWSCAsyncFor : implements IWSCAsyncFor, public CInterface, public CAsyncFo
         StringBuffer dbgheader, contentEncoding;
         bool chunked = false;
         size32_t read = 0;
+        bool srtn;
```
Comment (jakesmith) [inline common/thorhelper/thorsoapcall.cpp:2085]: minor: not sure what "srtn" is acronym for, maybe isClosed/socketClosed would be clearer?

**Review by jakesmith — COMMENTED** [review]
@mckellyln - looks good.

I think there's another case where same could happen in httpproxy.cpp, that would be worth changing at same time.
There's also a similar case in jbsocket.cpp (line 138) ..

File: common/thorhelper/thorsoapcall.cpp (hunk 1 above)
Comment (mckellyln) [reply common/thorhelper/thorsoapcall.cpp:2085]: I'm old - its just short for status return :-)  I can change it to sockClosed.

**Comment by mckellyln** [comment]
I have httpproxy.cpp change in this PR.  Yes I will now look at jbsocket.cpp

**Comment by mckellyln** [comment]
@jakesmith I've pushed a second commit for changes.  Lets see how smoke tests behave.

**Review by jakesmith — APPROVED** [review]
@mckellyln - looks good, please squash.

File: common/thorhelper/thorsoapcall.cpp [hunk 2]
```diff
@@ -2201,7 +2202,8 @@ class CWSCAsyncFor : implements IWSCAsyncFor, public CInterface, public CAsyncFo
             }
             if (read)
                 response.append(read,payload);
-            if (payloadsize) {  // read directly into response
+            if (payloadsize) // read directly into response
+            {
                 while (read<payloadsize) {
                     checkTimeLimitExceeded(&remainingMS);
                     checkRoxieAbortMonitor(master->roxieAbortMonitor);
```
Comment (ghalliday) [inline common/thorhelper/thorsoapcall.cpp:2209]: @mckellyln does the line below need a similar change? on line 2216

**Review by ghalliday — COMMENTED** [review]
@mckellyln one question.

File: common/thorhelper/thorsoapcall.cpp (hunk 2 above)
Comment (mckellyln) [reply common/thorhelper/thorsoapcall.cpp:2209]: I did not think so because the loop is checking for a specific size.

Comment (mckellyln) [reply common/thorhelper/thorsoapcall.cpp:2210]: Yes, updated code.

**Comment by mckellyln** [comment]
squashed

**Review by jakesmith — APPROVED** [review]
@mckellyln - looks good.

**Pull request merged**